src/OutputTransformer/ConsoleOutputTransformerFactory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phpcq\OutputTransformer;
+
+use Phpcq\PluginApi\Version10\OutputInterface;
+use Phpcq\PluginApi\Version10\OutputTransformerFactoryInterface;
+use Phpcq\PluginApi\Version10\OutputTransformerInterface;
+use Phpcq\PluginApi\Version10\ToolReportInterface;
+use Phpcq\PluginApi\Version10\Util\BufferedLineReader;
+
+final class ConsoleOutputTransformerFactory implements OutputTransformerFactoryInterface
+{
+    /**
+     * @var string
+     */
+    private $toolName;
+
+    /**
+     * ConsoleOutputTransformerFactory constructor.
+     *
+     * @param string $toolName
+     */
+    public function __construct(string $toolName)
+    {
+        $this->toolName = $toolName;
+    }
+
+    public function createFor(ToolReportInterface $report): OutputTransformerInterface
+    {
+        return new class ($report) implements OutputTransformerInterface {
+            /** @var ToolReportInterface */
+            private $report;
+            /** @var BufferedLineReader */
+            private $data;
+            /** @var string */
+            private $stdErr = '';
+            /** @var string */
+            private $stdOut ='';
+
+            /**
+             * Create a new instance.
+             */
+            public function __construct(ToolReportInterface $report)
+            {
+                $this->report = $report;
+                $this->data   = new BufferedLineReader();
+            }
+
+            public function write(string $data, int $channel): void
+            {
+                $this->data->push($data);
+                if (OutputInterface::CHANNEL_STDOUT === $channel) {
+                    $this->stdOut .= $data;
+                    return;
+                }
+                if (OutputInterface::CHANNEL_STDERR === $channel) {
+                    $this->stdErr .= $data;
+                    return;
+                }
+            }
+
+            public function finish(int $exitCode): void
+            {
+                $content = '';
+                while ($line = $this->data->fetch()) {
```

Comment:
There's no need to use the buffered line reader here. Collecting the output in a string parameter would be enough.

Reply:
You have in mind that the lines are trimmed by the buffered reader?
This fixes "ugly" line wrapping issues.

Reply:
Missed that point.